Foundation/Process.swift

```diff
@@ -978,4 +979,127 @@ private func posix(_ code: Int32) {
     default: fatalError("POSIX command failed with error: \(code)")
     }
 }
+
+#if os(Android)
+// Android doesn't provide posix_spawn APIs until recent API level, so we cannot
```

Comment:
Do you want to delegate to the real API if they exist?

Reply:
I don’t know if one can do that.

The availability of the symbol depends on the Android API level. By default, if nobody sets a different `__ANDROID_API__` while compiling, the latest API level is used. Swift sets API level 21, but Foundation doesn't set anything, so at the moment, Foundation is being compiled with access to every available symbol in the Android NDK (I'm not 100% sure about that, though, because I don’t know if the module cache remembers the setting from Swift). So the symbol will be available at compile time, but it might not be at runtime (and the binary will probably fail to start).

I don’t think `#available/@available` works for Android, which might be a good way, so I think this is the best I can come up now.

Reply:
I think that it makes sense to pass along the `__ANDROID_API__` when building for Android.  Using `posix_spawn` is preferable to rolling our own.

Reply:
I don't understand where you exactly want to pass `__ANDROID_API__`. Swift definitions are just booleans, and there's no support for something like `#if __ANDROID_API__ >= 21`, IIRC.

We might be able to do those checks in CoreFoundation, but again `__ANDROID_API__` is not a runtime check, but a "minimum deployment target" compilation check. When compiling you are deciding which is the minimal API you want to deploy to. As the code is written, we are supporting API levels before 26. If I use posix_spawn (and link against it), then we are not supporting anything before level 26.

That's my understanding, unless someone has another theory.